### PR TITLE
Use a better way to fix negotiation result

### PR DESCRIPTION
**Pull request by zackliu**
# Contributing to the Azure SDK

Use a better way to fix negotiation result and remove System.IdentityModel.Tokens.Jwt

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/src/Services/WebPubSubForSocketIOService.cs [hunk 1]
```diff
@@ -32,14 +30,18 @@ public WebPubSubForSocketIOService(Uri endpoint, AzureKeyCredential keyCredentia
...
+            _endpoint = endpoint;
             _hub = hub;
             _useConnectionStrings = false;
         }
 
         // For tests.
-        public WebPubSubForSocketIOService(WebPubSubServiceClient client)
+        public WebPubSubForSocketIOService(WebPubSubServiceClient client, Uri endpoint = null, string hub = null, bool useConnectionStrings = false)
         {
             _client = client;
```
Comment (vicancy) [inline sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/src/Services/WebPubSubForSocketIOService.cs:41]: we also have client.Endpoint in WebPubSubServiceClient, endpoint parameter is not needed?

File: sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/tests/SocketIOServiceTests.cs [hunk 2]
```diff
@@ -80,15 +80,16 @@ public void TestValidAadSchema(string endpoint, string host)
             Assert.Null(key);
         }
 
-        [Test]
-        public void TestNegotiateResultForAad()
+        [TestCase("https://sio-5kkfcgr2obqvm.webpubsub.azure.com/")]
+        [TestCase("https://sio-5kkfcgr2obqvm.webpubsub.azure.com")]
+        public void TestNegotiateResultForAad(string uri)
         {
             var token = "eyJhbGciOiJIUzI1NiIsImtpZCI6InMtZjZlMTVhZmItNjIxZS00OTc5LTgyZTgtN2FiMGQ4ZmIwMDM1IiwidHlwIjoiSldUIn0.eyJuYmYiOjE3MjcwNzAxODQsImV4cCI6MTcyNzA3MzcyNCwiaWF0IjoxNzI3MDcwMTg0LCJpc3MiOiJodHRwczovL3dlYnB1YnN1Yi5henVyZS5jb20iLCJhdWQiOiJodHRwczovL3Npby01a2tmY2dyMm9icXZtLndlYnB1YnN1Yi5henVyZS5jb20vY2xpZW50cy9zb2NrZXRpby9odWJzL2h1YiJ9.h3QkRTQ4";
```
Comment (vicancy) [inline sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/tests/SocketIOServiceTests.cs:87]: is this sensitive data?

File: sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/src/Services/WebPubSubForSocketIOService.cs [hunk 3]
```diff
@@ -32,14 +30,18 @@ public WebPubSubForSocketIOService(Uri endpoint, AzureKeyCredential keyCredentia
         public WebPubSubForSocketIOService(Uri endpoint, TokenCredential credential, string hub)
```
Comment (vicancy) [inline sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/src/Services/WebPubSubForSocketIOService.cs:30]: `:this(new WebPubSubServiceClient(endpoint, hub, credential), hub, false)`?

File: sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/tests/SocketIOServiceTests.cs (hunk 2 above)
Comment (zackliu) [reply sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/tests/SocketIOServiceTests.cs:87]: Already redacted the signature

File: sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/src/Services/WebPubSubForSocketIOService.cs (hunk 1 above)
Comment (zackliu) [reply sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO/src/Services/WebPubSubForSocketIOService.cs:32]: That's correct

**Review by vicancy — APPROVED** [review]

**Pull request merged**